src/network/bmproto.py

```diff
@@ -351,7 +351,7 @@ def _command_inv(self, extend_dandelion_stem=False):
             raise BMProtoExcessiveDataError()
 
         # ignore dinv if dandelion turned off
-        if extend_dandelion_stem and not state.dandelion_enabled:
+        if extend_dandelion_stem and not dandelion.instance.enabled:
```

Comment:
what we can do here is to move this check into the `Dandelion.hasHash` method. Then bmproto doesn't need to check.

Reply:
we can move into Dandelion.hasHash. but `hasHash` here used in some checks in iteration. i think that could not work same as here, like return direct True.

Reply:
The easiest solution is just to use the same approach in `if` as in `for`. It would result in dummy loops but my guess is that it's not a big performance hit. It could be optimised for performance later if needed.